Documentation/data-type-and-flow-control.md

```diff
@@ -0,0 +1,126 @@
+| offset/type=uint16 | 0 | 2 | 4 | 6 | 8 | 10 | 12 | 14 | 16 | 18 | 20 | 22 | 24 | 26 | 28 | 30 |
+
+Therefore, array in C corresponding to the buffer of a ```other/tensor``` becomes:
+```
+type buffer[dim4][dim3][dim2][dim1];
+```
+
+Note that in some context (such as properties of ```tensor_*``` plugins), dimensions are described in a colon-separated string that allows omitting unused dimensions:
+```
+dim1:dim2:dim3:dim4
+```
+If rank = 2 (dim3 = 1 and dim4 = 1), then, it can be expressed as well as:
+```
+dim1:dim2
+```
+
+Be careful! Colon-separated tensor dimension expression has the opposite order to the C-array type expression.
+
+## other/tensors
+
+```other/tensors``` is defined to have multiple instances of ```other/tensor``` in a buffer of a single stream path. Compared to having multiple streams (thus multiple pads) with ```other/tensor``` that goes to or comes from a single element, having a single stream with ```other/tensors``` has the following advantages:
+
+- ```tensor_filter```, which is the main engine to communicate deep neural network frameworks and models, becomes simple and robust. With neural network models requiring multiple input tensors, if the input tensor streams are not fully synchronized, we need to somehow synchronize them and provide all input tensors at the same time to the model. Hereby, being fully synchronized means that the streams should provide new data simultaneously, which requires to have the exactly same framerate and timing, which is mostly impossible. With ```other/tensors``` and single input and output streams for ```tensor_filter```, we can delegate the responsibilities of synchronization to GStreamer and its basic plugins, who are extremely good at such tasks.
+- During transmissions on a stream pipeline, passing through various stream filters, we can guarantee that the same set of input tensors are being processed without the worries of synchronizations after the point of merging or muxing.
+
+The GStreamer pad capability of ```other/tensors``` is as follows:
+```
+other/tensors
+    num_tensors = (int) [1, 16]  # GST_MAX_MEMCHUNK_PER_BUFFER
+    framerate = (fraction) [0/1, 2147483647/1]
+    types = (string) Typestrings
+    dimensions = (string) Dimensions
+
+Typestrings = (string) Typestring
+            | (string) TypeString, TypeStrings
+Typestring = (string) { float32, float64, int64, uint64, int32, uint32, int16, uint16, int8, uint8 }
+Dimensions = (string) Dimension
+           | (string) Dimension, Dimensions
+Dimension = (string) [1-65535]:[1-65535]:[1-65535]:[1-65535]
+```
+
+The buffer of ```other/tensors``` streams have multiple memory chunks. Each memory chunk represents one single tensor in the buffer format of ```other/tensor```. With default configurations of Gstreamer 1.0, the maximum allowed number of memory chunks in a buffer is **16**. Thus, with such configurations of Gstreamer 1.0, ```other/tensors``` may include up to **16** ```other/tensor```.
+
+
+## other/tensorsave

```

Comment:
Discussion: `o/tensorsave` is necessary?

Reply:
`other/tensorsave` has not been implemented. Marked as `TBU`